app/code/Magento/InventoryApi/Api/Data/ReservationInterface.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+use Magento\InventoryApi\Api\Data\ReservationExtensionInterface;
+
+/**
+ * The entity responsible for reservations, created to keep inventory amount (product quantity) up-to-date.
+ * It is created to have a state between order creation and inventory deduction (deduction of specific SourceItems).
+ *
+ * Reservations are designed to be immutable entities.
+ *
+ * @api
+ */
+interface ReservationInterface extends ExtensibleDataInterface
```

Comment:
usually, we extend from ExtensibleDataInterface in case we want to provide an ability to add custom attributes to some particular entity using methods get/setExtensibleAttributes 
you can read about this here - http://devdocs.magento.com/guides/v2.0/extension-dev-guide/extension_attributes/adding-attributes.html

We also use this functionality for entities being exposed outside, like web api.
don't think we need to extend from ExtensibleDataInterface in this particular case